frontend/src/components/Timetable/index.tsx

```diff
@@ -0,0 +1,123 @@
+import Flex from '../Layout/Flex';
+import Wrapper from '../Layout/Wrapper';
+import { useState } from 'react';
+import useTimeSelection from '@/hooks/TimeTable/useTimeSelection';
+import { useTheme } from '@emotion/react';
+import Text from '../Text';
+import { getDayOfWeek } from '@/utils/Calendar/getDayofWeek';
+import * as S from './Timetable.styled';
+import generateTimeList from '@/utils/Calendar/generateTimeList';
+import Button from '../Button';
+import { Field } from '@/types/field';
+
+interface TimetableProps {
+  availableDates: Set<string>;
+  startTime: string;
+  endTime: string;
+  session: string | undefined;
+  userAvailability: {
+    userName: Field<string>;
+    selectedTimes: Field<Set<string>>;
+    userAvailabilitySubmit: (session: string | undefined) => Promise<unknown>;
+  };
+}
+
+const Timetable = ({
+  availableDates,
+  startTime,
+  endTime,
+  session,
+  userAvailability,
+}: TimetableProps) => {
+  const startTimeInMinutes = startTime.split(':').reduce((acc, time) => acc * 60 + +time, 0);
+  const endTimeInMinutes = endTime.split(':').reduce((acc, time) => acc * 60 + +time, 0);
+  const interval = 30;
+
+  const timeList = [
+    { timeText: 'Dates', isHour: false },
+    ...generateTimeList({ startTimeInMinutes, endTimeInMinutes, interval }),
+  ];
+
+  const { onMouseDown, onMouseEnter, onMouseUp, onMouseLeave } = useTimeSelection({
+    selectedTimes: userAvailability.selectedTimes.value,
+    setSelectedTimes: userAvailability.selectedTimes.set,
+    time: '',
+  });
+
+  const theme = useTheme();
+
+  return (
+    <Wrapper
+      maxWidth={800}
+      backgroundColor={theme.colors.background}
+      paddingTop="var(--padding-9)"
+      paddingRight="var(--padding-9)"
+      paddingBottom="var(--padding-9)"
+      paddingLeft="var(--padding-9)"
+      borderRadius="var(--radius-6)"
+    >
+      <Flex direction="column" gap="var(--gap-6)">
+        <S.TimetableHeader>
+          <Flex direction="column" gap="var(--gap-4)">
+            <Text variant="h2" color="text">
+              {userAvailability.userName.value}의 시간 등록하기
```

Comment:
상위에서 userAvailability 객체로 풀어서 적용하면 더 가독성이 올라갈 것 같네요!
```suggestion
const {userName,userAvailabilitySubmit }  = userAvailability;
...
{userName.value}의 시간 등록하기
```